ui/components/editor-submit-dialog.vue

```diff
@@ -5,56 +5,123 @@
+        <a
+          href="/fixture-editor"
+          class="button secondary"
+          @click.prevent="$emit(`reset`)">Create another fixture</a>
+        <app-download-button :showHelp="false" :editor-fixtures="submit.sendObject" />
+        <a :href="pullRequestUrl" class="button primary" target="_blank">See pull request</a>
       </div>
     </div>
 
     <div v-else-if="submit.state === `error`">
-      <span>Unfortunately, there was an error while uploading. Please copy the following data and <a href="https://github.com/OpenLightingProject/open-fixture-library/issues/new" target="_blank">manually submit them to GitHub</a>.</span>
+      Unfortunately, there was an error while uploading. Please copy the following data and
+      <a
+        href="https://github.com/OpenLightingProject/open-fixture-library/issues/new"
+        target="_blank">manually submit them to GitHub</a>.
 
-      <textarea v-model="submit.rawData" readonly />
+      <textarea v-model="rawData" readonly />
 
       <div class="button-bar right">
-        <nuxt-link to="/" class="button secondary">Back to homepage</nuxt-link>
-        <a href="https://github.com/OpenLightingProject/open-fixture-library/issues/new" class="button primary" target="_blank">Submit manually</a>
+        <a href="#cancel" class="button secondary" @click.prevent="onCancel">Close</a>
+        <a
+          href="https://github.com/OpenLightingProject/open-fixture-library/issues/new"
+          class="button primary"
+          target="_blank">Submit manually</a>
       </div>
     </div>
 
-    <!-- <div v-else-if="submit.state === `invalid`">
-    Unfortunately, the fixture you uploaded was invalid. Please correct the following mistakes before trying again.
-    <textarea v-model="submit.mistakes" readonly />
-    <textarea v-model="submit.rawData" readonly />
-    <div class="button-bar right">
-    <button class="primary" data-action="home">Back to homepage</button>
-    </div>
-    </div> -->
-
   </app-a11y-dialog>
 </template>
 
+<style lang="scss">
+#submit-dialog {
+  // used for the download button to be able to show
+  // the drop down over the dialog
+  overflow: unset;
+}
+</style>
+
 <script>
 import a11yDialogVue from '~/components/a11y-dialog.vue';
+import downloadButtonVue from '~/components/download-button.vue';
+import { clone } from '~/assets/scripts/editor-utils.mjs';
 
 export default {
   components: {
-    'app-a11y-dialog': a11yDialogVue
+    'app-a11y-dialog': a11yDialogVue,
+    'app-download-button': downloadButtonVue
   },
   props: {
     submit: {
       type: Object,
       required: true
     }
   },
+  data() {
+    return {
+      error: null,
+      pullRequestUrl: null,
+      downloadFileName: null,
+      downloadFileUrl: null,
+      validationErrors: [],
+      validationWarnings: [],
+      fixtureJson: null
+    };
+  },
   computed: {
     title() {
```

Comment:
Please order the states the same way as in the markup – and maybe refactor this method using a map.